modules/apps/object/object-web/src/main/java/com/liferay/object/web/internal/object/entries/frontend/data/set/provider/SystemRelatedModelsFDSDataProvider.java

```diff
@@ -83,35 +91,46 @@ public List<RelatedModel> getItems(
 		long objectEntryId = ParamUtil.getLong(
 			httpServletRequest, "objectEntryId");
 
+		User user = _userLocalService.getUser(PrincipalThreadLocal.getUserId());
+
 		return TransformUtil.transform(
 			(List<BaseModel<?>>)objectRelatedModelsProvider.getRelatedModels(
 				objectScopeProvider.getGroupId(httpServletRequest),
 				objectRelationshipId, objectEntryId, fdsKeywords.getKeywords(),
 				fdsPagination.getStartPosition(),
 				fdsPagination.getEndPosition()),
 			relatedModel -> {
-				String objectFieldDBColumnName =
-					objectDefinition.getPKObjectFieldDBColumnName();
-
-				if (objectDefinition.getTitleObjectFieldId() > 0) {
-					ObjectField objectField =
-						_objectFieldLocalService.getObjectField(
-							objectDefinition.getTitleObjectFieldId());
-
-					objectFieldDBColumnName = objectField.getDBColumnName();
+				ObjectField titleObjectField =
+					_objectFieldLocalService.fetchObjectField(
+						objectDefinition.getTitleObjectFieldId());
+
+				if (titleObjectField == null) {
+					titleObjectField =
+						_objectFieldLocalService.fetchObjectField(
+							objectDefinition.getObjectDefinitionId(), "id");
 				}
 
-				Map<String, Object> modelAttributes =
-					relatedModel.getModelAttributes();
+				Map<String, Object> values = ObjectEntryDTOConverterUtil.toDTO(
+					relatedModel, _dtoConverterRegistry,
+					objectDefinition.getName(),
+					_systemObjectDefinitionManagerRegistry, user);
 
-				Object value = modelAttributes.get(objectFieldDBColumnName);
+				if (!StringUtil.equals(
+						objectDefinition.getName(), "CPDefinition")) {
+
+					values.put("createDate", values.get("dateCreated"));
+					values.put("modifiedDate", values.get("dateModified"));
+				}
```

Comment:
@seltong why do we need this condition? Isn't [this code](https://github.com/liferay-objects/liferay-portal/blob/133d0457a11475c773ebadb443df8b2270c048e8/modules/apps/object/object-api/src/main/java/com/liferay/object/entry/util/ObjectEntryValuesUtil.java#L45C1-L47) enough?